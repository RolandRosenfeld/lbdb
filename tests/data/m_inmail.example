foo@example.com	Bb2 Bb2	2022-01-05 12:00
bar@example.com	"Ee5 Ee5"	2022-01-01 12:00
baz@example.com	Aa1 Aa1	2022-01-04 12:00
fofo@example.com	Dd4 Dd4	2022-01-02 12:00
baba@example.com	Cc3 Cc3	2022-01-03 12:00
bzbz@example.com	"Ff6 Ff6"	2022-01-06 12:00
baz@example.com	Ac1 Ac1	2022-01-07 12:00
baz@example.com	Ab1 Ab1	2022-01-08 12:00
last1@example.de	First Last1	2022-01-09 12:00
last2@example.de	First Last2	2022-01-10 12:00
