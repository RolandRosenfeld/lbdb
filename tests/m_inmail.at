dnl Copyright (c) 2022-2025 Roland Rosenfeld <roland@spinnaker.de>
dnl
dnl This program is free software; you can redistribute it and/or modify
dnl it under the terms of the GNU General Public License as published by
dnl the Free Software Foundation; either version 2 of the License, or
dnl (at your option) any later version.
dnl
dnl This program is distributed in the hope that it will be useful,
dnl but WITHOUT ANY WARRANTY; without even the implied warranty of
dnl MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
dnl GNU General Public License for more details.
dnl
dnl You should have received a copy of the GNU General Public License along
dnl with this program; if not, see <https://www.gnu.org/licenses/>.

AT_BANNER([Test m_inmail/fetchmail])

AT_SETUP(lbdb-fetchmail)
AT_CHECK([cat $abs_srcdir/data/m_inmail.utf8.expected1 > expout])
AT_CHECK([cat $abs_srcdir/data/testmail1 > testmail1])
AT_CHECK([lbdb-fetchaddr -f m_inmail.utf8 -d "NO DATE" < testmail1 && \
 	  lbdb-fetchaddr -f m_inmail.utf8 -d "NO DATE" \
			 -x x-header1:x-header2:x-header3:x-header4 < testmail1 && \
	  cat m_inmail.utf8], [], [expout])
AT_CLEANUP

AT_SETUP(m_inmail)
AT_CHECK([cat $abs_srcdir/data/m_inmail.utf8.expected2 > expout])
AT_CHECK([cat $abs_srcdir/data/testmail1 > testmail1])
AT_CHECK([lbdb-fetchaddr -f m_inmail.utf8 -d "NO DATE" < testmail1 && \
 	  lbdb-fetchaddr -f m_inmail.utf8 -d "NO DATE" < testmail1])
AT_CHECK([env LBDB_OVERRIDE_METHODS=m_inmail \
	      OVERRIDE_M_INMAIL_DB=m_inmail.utf8 \
	      OVERRIDE_LIMITDATE="0000-00-00_00:00" \
	      OVERRIDE_LIMITCOUNT=0 \
	      OVERRIDE_SORT_OUTPUT=no \
          lbdbq GÃ¼nther], [], [expout])
AT_CLEANUP

AT_SETUP(m_inmail SORT=address)
AT_CHECK([cat $abs_srcdir/data/m_inmail.example > m_inmail.utf-8])
AT_CHECK([cat $abs_srcdir/data/m_inmail.sortaddress.expected > expout])
AT_CHECK([env LBDB_OVERRIDE_METHODS=m_inmail \
	      OVERRIDE_M_INMAIL_DB=m_inmail.utf-8 \
	      OVERRIDE_LIMITDATE="0000-00-00_00:00" \
	      OVERRIDE_LIMITCOUNT=0 \
	      OVERRIDE_SORT_OUTPUT=address \
          lbdbq example.com], [], [expout])
AT_CLEANUP

AT_SETUP(m_inmail SORT=name)
AT_CHECK([cat $abs_srcdir/data/m_inmail.example > m_inmail.utf-8])
AT_CHECK([cat $abs_srcdir/data/m_inmail.sortname.expected > expout])
AT_CHECK([env LBDB_OVERRIDE_METHODS=m_inmail \
	      OVERRIDE_M_INMAIL_DB=m_inmail.utf-8 \
	      OVERRIDE_LIMITDATE="0000-00-00_00:00" \
	      OVERRIDE_LIMITCOUNT=0 \
	      OVERRIDE_SORT_OUTPUT=name \
          lbdbq example.com], [], [expout])
AT_CLEANUP

AT_SETUP(m_inmail SORT=comment)
AT_CHECK([cat $abs_srcdir/data/m_inmail.example > m_inmail.utf-8])
AT_CHECK([cat $abs_srcdir/data/m_inmail.sortcomment.expected > expout])
AT_CHECK([env LBDB_OVERRIDE_METHODS=m_inmail \
	      OVERRIDE_M_INMAIL_DB=m_inmail.utf-8 \
	      OVERRIDE_LIMITDATE="0000-00-00_00:00" \
	      OVERRIDE_LIMITCOUNT=0 \
	      OVERRIDE_SORT_OUTPUT=comment \
          lbdbq example.com], [], [expout])
AT_CLEANUP

AT_SETUP(m_inmail SORT=reverse_comment)
AT_CHECK([cat $abs_srcdir/data/m_inmail.example > m_inmail.utf-8])
AT_CHECK([cat $abs_srcdir/data/m_inmail.sortreverse_comment.expected > expout])
AT_CHECK([env LBDB_OVERRIDE_METHODS=m_inmail \
	      OVERRIDE_M_INMAIL_DB=m_inmail.utf-8 \
	      OVERRIDE_LIMITDATE="0000-00-00_00:00" \
	      OVERRIDE_LIMITCOUNT=0 \
	      OVERRIDE_SORT_OUTPUT=reverse_comment \
          lbdbq example.com], [], [expout])
AT_CLEANUP

AT_SETUP(m_inmail whitespace)
AT_CHECK([cat $abs_srcdir/data/m_inmail.example > m_inmail.utf-8])
AT_CHECK([cat $abs_srcdir/data/m_inmail.whitespace.expected > expout])
AT_CHECK([env LBDB_OVERRIDE_METHODS=m_inmail \
	      OVERRIDE_M_INMAIL_DB=m_inmail.utf-8 \
	      OVERRIDE_LIMITDATE="0000-00-00_00:00" \
	      OVERRIDE_LIMITCOUNT=0 \
	      OVERRIDE_SORT_OUTPUT=no \
	lbdbq first last1], [], [expout])
AT_CLEANUP

AT_SETUP(m_inmail whitespace quote)
AT_CHECK([cat $abs_srcdir/data/m_inmail.example > m_inmail.utf-8])
AT_CHECK([cat $abs_srcdir/data/m_inmail.whitespace.expected > expout])
AT_CHECK([env LBDB_OVERRIDE_METHODS=m_inmail \
	      OVERRIDE_M_INMAIL_DB=m_inmail.utf-8 \
	      OVERRIDE_LIMITDATE="0000-00-00_00:00" \
	      OVERRIDE_LIMITCOUNT=0 \
	      OVERRIDE_SORT_OUTPUT=no \
	lbdbq "first last1"], [], [expout])
AT_CLEANUP
