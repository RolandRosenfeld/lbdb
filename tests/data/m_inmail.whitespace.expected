lbdbq: 1 matches
last1@example.de	First Last1	2022-01-09 12:00
